### ISSUE TITLE
unusedArguments should implicitly bypass required initializers

### ISSUE THREAD

**Issue by RabugenTom** [nicklockwood/SwiftFormat#99]
Hi Nick,

I suggest that `unusedArguments` should implicitly bypass required initializers (and maybe be optionally reactivated in these cases). Otherwise, SwiftFormat may break code when used with default settings.

As an example, by default, SwiftFormat breaks simple [ObjectMapper](https://github.com/Hearst-DD/ObjectMapper) implementations where
```swift
required init(map: Map) {
}
```
is formatted as 
```swift
required init(map _: Map) {
}
```
which is no more the expected signature for the `Mappable` protocol. The trivial implementation is the most frequent case, and it's required so one has to either make a dummy reference to `map` or use the `--stripunusedargs unnamed-only` option globally. I think this is not optimal.

**Comment by nicklockwood** [comment]
@rabugentom it's my understanding that

    init(map: Map)

And

    init(map _: Map)

Are the same method signature as far as Swift is concerned. Do you actually have code that is failing to compile after this transformation?

**Comment by RabugenTom** [comment]
Sorry, I made an error while writing the code. This is a failable initializer:
````swift 
required init?(map: Map) {
}
````
is formatted as 
````swift 
required init?(_: Map) {
}
````
This is a bug caused by the `?`. If one removes it, one falls back to the first case where the signature is indeed not changed.

**Comment by nicklockwood** [comment]
@rabugentom ah, thanks. That does sound like a bug then - I'll investigate.

**Comment by nicklockwood** [comment]
Fixed in 0.25.1. Thanks!
